### PR TITLE
[unit-tests-only] Run minimum pipelines for unit tests only PRs

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
if the PR title has `unit-tests-only` then only the minimum pipelines will be executed in CI

Minimum Pipelines for unit-tests-only: 
- lint-test
- stop-recent-builds
- unit-and-integration-tests
- check-starlark

## Related Issue
- Fixes https://github.com/owncloud/web/issues/5421

## Motivation and Context
Optimize the CI build duration

## How Has This Been Tested?
- test environment: CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -727,7 +727,11 @@ def main(ctx):
     return pipelines + deploys + checkStarlark()
 
 def beforePipelines(ctx):
-    return yarnlint() + checkForRecentBuilds(ctx) + changelog(ctx) + website(ctx) + cacheOcisPipeline(ctx)
+    title = ctx.build.title.lower()
+    if "[unit-test-only]" in title or "[unit-tests-only]" in title:
```
Comment (phil-davis) [inline .drone.star:731]:
```suggestion
    if "unit-test-only" in title or "unit-tests-only" in title:
```
We don't usually require the `[]` - that is just a habit that we are in when writing PR titles.

Comment (saw-jan) [reply .drone.star:731]: Yeah, I was making it strict. As my PR title has `unit-tests-only` in description part `[unit-tests] Run minimum pipelines for unit-tests-only PRs`, this is also treated as unit-test-only and runs minimum pipelines.

Comment (phil-davis) [reply .drone.star:731]: It's tricky to know what to do exactly - if someone puts `(unit-tests-only)` then that should also be OK.

**Review by phil-davis — APPROVED** [review]
